src/linker/Linker/MessageContainer.cs

```diff
@@ -70,6 +70,12 @@ public static MessageContainer CreateWarningMessage (LinkContext context, string
 			if (context.IsWarningSuppressed (code, origin))
 				return Empty;
 
+			origin.TryGetSourceInfo ();
+			if (subcategory == "Unrecognized reflection pattern" && origin.FileName == null) {
```

Comment:
This should probably happen for every warning. If we can't figure out the filename but we know which member it belongs to, we should report the member. This is not specific to the reflection patterns.